### PR TITLE
fix(clearRefinements): reset page to 0

### PR DESCRIPTION
**Pull request by eunjae-lee**
**Summary**

`clearRefinements` is not resetting the page to 0.
This is not an expected behaviour and brings a confusion to users.
When everything is reset, the page should be reset to 0.

**Result**

Now it resets page to 0.

### CODE REVIEW

File: src/connectors/clear-refinements/__tests__/connectClearRefinements-test.js [hunk 1]
```diff
@@ -517,5 +517,26 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/clear-refin
...
+      widget.init({
+        helper,
+        state: helper.state,
+        createURL: () => '#',
+      });
+      const clearRefinements = rendering.mock.calls[0][0].refine;
+
+      helper.setPage(2);
+      clearRefinements();
+      expect(helper.getPage()).toBe(0);
```
Comment (samouss) [inline src/connectors/clear-refinements/__tests__/connectClearRefinements-test.js]: Prefer the usage of `helper.state.page`. We never know we might remove those getters at some point.

File: src/lib/utils/clearRefinements.js [hunk 2]
```diff
@@ -21,6 +21,7 @@ function clearRefinements({ helper, attributesToClear = [] }) {
   if (attributesToClear.indexOf('query') !== -1) {
     finalState = finalState.setQuery('');
   }
+  finalState = finalState.setPage(0);
```
Comment (samouss) [inline src/lib/utils/clearRefinements.js]: We could have inlined the call into the first assignment.

Comment (eunjae-lee) [reply src/lib/utils/clearRefinements.js]: 1fc7375

File: src/connectors/clear-refinements/__tests__/connectClearRefinements-test.js (hunk 1 above)
Comment (eunjae-lee) [reply src/connectors/clear-refinements/__tests__/connectClearRefinements-test.js]: 43f9c39

**Pull request merged**